lib/rules/sort-keys.js

```diff
@@ -159,8 +166,36 @@ module.exports = {
                 const numKeys = stack.numKeys;
                 const thisName = getPropertyName(node);
 
+                // Get tokens between current node and previous node
+                const tokens = stack.prevNode && sourceCode
+                    .getFirstTokensBetween(stack.prevNode, node, { skip: 0, includeComments: true });
+
+                let isBlankLineBetweenNodes = false;
+
+                if (tokens) {
+
+                    // check blank line between tokens
+                    tokens.forEach((token, index) => {
+                        const previousToken = tokens[index - 1];
+
+                        if (previousToken && (token.loc.start.line - previousToken.loc.end.line > 1)) {
+                            isBlankLineBetweenNodes = true;
+                        }
+                    });
+
+                    // check blank line between current node and last token
+                    if (!isBlankLineBetweenNodes && (node.loc.start.line - tokens[tokens.length - 1].loc.start.line > 1)) {
+                        isBlankLineBetweenNodes = true;
+                    }
+                }
+
                 if (thisName !== null) {
                     stack.prevName = thisName;
+                    stack.prevNode = node;
+                }
```

Comment:
I think this would be a false negative, since the properties are not line separated:

```js
/* eslint sort-keys: ["error", "asc", { allowLineSeparatedGroups: true } ] */

var obj = {
  b,
  [foo()]: [
  // ↓ this blank is inside a property and therefore should not count

  ],
  a
}

```

Reply:
Fixed in https://github.com/eslint/eslint/pull/16138/commits/3c441017942220f1b8772d72767cc3506977fdc5